admin-dev/themes/new-theme/js/pages/product/components/catalog-price-rule/catalog-price-rule-list.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+import {getCatalogRules} from '@pages/product/services/catalog-rule-service';
+import {EventEmitter} from 'events';
+import ProductMap from '@pages/product/product-map';
+
+const CatalogPriceRuleMap = ProductMap.catalogPriceRule;
+
+export default class CatalogPriceRuleList {
+  private eventEmitter: EventEmitter;
+
+  private listContainer: HTMLElement
+
+  constructor() {
+    this.listContainer = document.querySelector(CatalogPriceRuleMap.listContainer) as HTMLElement;
+    this.eventEmitter = window.prestashop.instance.eventEmitter;
+  }
+
+  public renderList(): void {
+    const {listFields} = CatalogPriceRuleMap;
+    const tbody = this.listContainer.querySelector(`${CatalogPriceRuleMap.listContainer} tbody`) as HTMLElement;
+    const trTemplateContainer = this.listContainer.querySelector(CatalogPriceRuleMap.listRowTemplate) as HTMLScriptElement;
+    const rowContainer = document.querySelector(CatalogPriceRuleMap.catalogPriceRuleRow) as HTMLElement;
```

Comment:
Please don't cast, use the generic

Example:
`document.querySelector<HTMLElement>('.yolo')`

Then deal with it!